commcare_connect/opportunity/visit_import.py

```diff
@@ -88,10 +90,31 @@ def _bulk_update_visit_status(opportunity: Opportunity, dataset: Dataset):
 
             UserVisit.objects.bulk_update(to_update, fields=["status"])
             missing_visits |= set(visit_batch) - seen_visits
+            update_payment_accrued(opportunity, users={visit.user_id for visit in visits})
 
     return VisitImportStatus(seen_visits, missing_visits)
 
 
+def update_payment_accrued(opportunity: Opportunity, users):
+    payment_units = opportunity.paymentunit_set.all()
+    for user in users:
+        user_visits = UserVisit.objects.filter(
+            opportunity=opportunity, user=user, status=VisitValidationStatus.approved
+        ).order_by("entity_id")
+        access = OpportunityAccess.objects.get(user=user, opportunity=opportunity)
+        payment_accrued = 0
+        for payment_unit in payment_units:
+            payment_unit_deliver_units = {deliver_unit.id for deliver_unit in payment_unit.deliver_units.all()}
+            for entity_id, visits in itertools.groupby(user_visits, key=lambda x: x.entity_id):
+                deliver_units = {v.deliver_unit.id for v in visits}
+                if payment_unit_deliver_units.issubset(deliver_units) or deliver_units.issubset(
```

Comment:
this logic seems incorrect to me. Specifically if a payment unit contains two deliver units and the user has done one of the `deliver_units.issubset(payment_deliver_units)` will be `True` but they will not have completed the payment unit (since they only did half of it). I think you want to check only the first condition (that the payment is a subset of what is done), but I might be missing an edge case you intended.